src/components/dialog.tsx

```diff
@@ -0,0 +1,490 @@
+              </Grid>
+              <Grid item xs={6}>
+                <TextField id="title" label="Title" variant="outlined" value = {bookData?.title} onChange = {(e) => setBookData({...bookData, title: e.target.value}) } fullWidth required />
+              </Grid>
+              <Grid item xs={6}>
+                <TextField id="subtitle" label="Subtitle" variant="outlined" value = {bookData?.subtitle} onChange = {(e) => setBookData({...bookData, subtitle: e.target.value}) } fullWidth />
+              </Grid>
+              <Grid item xs={6}>
+                <Autocomplete
+                  isOptionEqualToValue={(option, value) => option.value === value.value}
+                  disablePortal
+                  id="combo-box-language"
+                  options={languageOptions}
+                  onChange = {(e,item) => { setBookData({...bookData, language: Number(item?.value)}) }}
+                  renderInput={(params) => <TextField {...params} label="Language" required/>}
+                />
+              </Grid>
+              {/* Authors Dynamic Input */}
+              {objectAuthors.map((item, index) => (
+                <>
+                  <Grid item xs={6}>
+                    <TextField label={`Author ${index + 1}`} variant="outlined" value = {item.authorName} onChange={(event) => handleChangeAuthor(event, index)} fullWidth /> 
+                  </Grid>
+                  <Grid item xs={6}>
+                    {objectAuthors.length > 1 && (
+                      <IconButton aria-label="Delete Author" onClick={() => handleDeleteInput(index, setObjectAuthors, objectAuthors)}>
+                          <DeleteIcon/>
+                      </IconButton>
+                    )}
+                    {index === objectAuthors.length - 1 && (
+                      <IconButton aria-label="Add Author" onClick={() => handleAddInputAuthor()}>
+                          <AddCircleIcon />
+                      </IconButton>
+                    )}
+                  </Grid>
+                </>
+              ))}
+              <Grid item xs={6}>
+                <LocalizationProvider dateAdapter={AdapterDayjs}>
+                <DatePicker 
+                  label="Publish Date" 
+                  value = {textPublishDates} 
+                  onChange = {(e) => setTextPublishDates(e)}    
+                  slotProps={{
+                  textField: {
+                    id: 'publishDates',
+                  },
+                  }}/>
+                </LocalizationProvider>
+              </Grid>
+              <Grid item xs={6}>
+                <TextField type="number" id="numberOfPages" label="Number Of Pages" variant="outlined" value = { bookData?.numberOfPages } onChange = {(e) => setBookData({...bookData, numberOfPages: e.target.value}) } fullWidth /> 
```

Comment:
Aquí salió error en vercel, porque e.target.value es string y el type es number
`Type error: Type 'string' is not assignable to type 'number'.`

Reply:
Ah ya,deja lo arreglo

Reply:
Ya quedo @adlaiCarmona

Reply:
Deja arreglo el conflicto tambien